### PR TITLE
homework5

### PR DESCRIPTION
**Pull request by l0stdream**

### CODE REVIEW

File: src/homework5/Runner.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    public static int countOccurance(List<String> words, String targetWord) {
+        return Collections.frequency(words, targetWord);
+    }
+
+    public static <T> List<T> toList(T[] words) {
+        return Arrays.asList(words);
+    }
+
+    public static List<Integer> findUnique(List<Integer> numbers) {
+        return new ArrayList<>(new HashSet<Integer>(numbers));
```
Comment (shaitanaleksander) [inline src/homework5/Runner.java:16]: we don't use tools that we have not learn

File: src/homework5/Runner.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public static void calcOccurance(List<String> words) {
+        Set<String> uniqueWords = new HashSet<>(words);
```
Comment (shaitanaleksander) [inline src/homework5/Runner.java:20]: same here

File: src/homework5/Runner.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        return wordsFrequence;
+    }
+
+
+    public static LinkedList<Map<String, Integer>> findOccuranceWithMapsInList(List<String> words) {
+        HashMap<String, Integer> wordsAndOccurance = new HashMap<>();
+        LinkedList<Map<String, Integer>> result = new LinkedList<>();
+        for (String currentWord : words) {
+            wordsAndOccurance.put(currentWord, Collections.frequency(words, currentWord));
+        }
```
Comment (shaitanaleksander) [inline src/homework5/Runner.java:51]: need to be rewritten with out using hashMap

File: src/homework5/phonebook/Phonebook.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    public Record find(String name) {
+        for (Record currentRecord : this.phoneRecords) {
+            if (currentRecord.getName().equals(name))
+                return currentRecord;
+        }
+        return null;
+    }
+
+    public LinkedList<Record> findAll(String name) {
```
Comment (shaitanaleksander) [inline src/homework5/phonebook/Phonebook.java:25]: it is better to return List

File: src/homework5/phonebook/Phonebook.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        LinkedList<Record> matches = new LinkedList<>();
+        for (Record currentRecord : phoneRecords) {
+            if (currentRecord.getName().equals(name)) {
+                matches.add(currentRecord);
+            }
+        }
+        return matches.isEmpty() ? null : matches;
```
Comment (shaitanaleksander) [inline src/homework5/phonebook/Phonebook.java:32]: why not return an empty list instead of null?

**Review by shaitanaleksander — APPROVED** [review]

**Comment by l0stdream** [comment]
Corrected in another

**Pull request closed without merging**